frontend/packages/ceph-storage-plugin/src/components/kms-config/kms-config.tsx

```diff
@@ -179,15 +181,22 @@ export const KMSConfigure: React.FC<KMSConfigureProps> = ({ state, dispatch, mod
           validated={isValid(kms.token.valid)}
           isRequired
         >
-          <TextInput
-            value={kms.token.value}
-            onChange={setToken}
-            type="password"
-            id="kms-token"
-            name="kms-token"
-            isRequired
-            validated={isValid(kms.token.valid)}
-          />
+          <div className="ocs-install-kms__form-token">
+            <TextInput
+              value={kms.token.value}
+              onChange={setToken}
+              type={reveal ? 'text' : 'password'}
+              id="kms-token"
+              name="kms-token"
+              isRequired
+              validated={isValid(kms.token.valid)}
+            />
+            <Tooltip content={reveal ? 'Hide token' : 'Reveal token'}>
+              <Button variant="control" onClick={() => setReveal(!reveal)}>
+                {reveal ? <EyeSlashIcon /> : <EyeIcon />}
+              </Button>
+            </Tooltip>
+          </div>
```

Comment:
Instead of wrapping in div , you should consider `InputGroup` that shows a semantic connection of text input and button